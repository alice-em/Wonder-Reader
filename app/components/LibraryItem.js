import PropTypes from 'prop-types';
import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

import { FaFolderO, FaFileArchiveO, FaPercent } from 'react-icons/lib/fa';
// import Bookmark from './Bookmark.js';

const iconSize = 20;

const styles = {
  bbb: {
    color: '#bbb'
  },
  percent: {
    color: '#999',
    fontSize: '14px'
  },
  TableRow: {
    cursor: 'pointer',
    fontFamily: 'Montserrat',
    fontSize: '20px'
  },
  three33: {
    color: '#333'
  },
  wide: {
    width: '10px'
  }
};

const CellBaseName = ({ basename }) => (
  <TableCell style={styles.three33}>{basename}</TableCell>
);

const CellDirname = ({ dirname }) => (
  <TableCell numeric style={styles.bbb}>
    {dirname}
  </TableCell>
);

const CellIcon = ({ icon }) => (
  <TableCell numeric padding="checkbox" style={styles.wide}>
    {icon}
  </TableCell>
);

const CellPercent = () => (
  <TableCell padding="checkbox" style={styles.wide}>
    <FaPercent style={styles.percent} />
  </TableCell>
);

const LibraryItem = ({ basename, dirname, id, isDirectory, onRowClick }) => {
  const icon = isDirectory ? (
    <FaFolderO size={iconSize} />
  ) : (
    <FaFileArchiveO size={iconSize} />
  );

  return (
    <TableRow
      className="library-item"
      key={id}
      onClick={onRowClick}
      style={styles.TableRow}
    >
      <CellIcon icon={icon} />
      <CellBaseName basename={basename} />
      <CellDirname dirname={dirname} />
      <CellPercent />
    </TableRow>
  );
};

LibraryItem.propTypes = {
  basename: PropTypes.string.isRequired,
  dirname: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  isDirectory: PropTypes.bool.isRequired,
  onRowClick: PropTypes.func.isRequired
};

export default LibraryItem;
