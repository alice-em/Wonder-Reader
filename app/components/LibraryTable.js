import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableHead from '@material-ui/core/TableHead';

import LibraryItem from './LibraryItem';
import LibraryTableHead from './LibraryTableHead';

class LibraryTable extends Component {
  renderLibraryItem = content => {
    const {
      basename,
      dirname,
      id,
      isDirectory
    } = content;
    const { onContentClick } = this.props;

    return (
      <LibraryItem
        basename={basename}
        dirname={dirname}
        id={id}
        isDirectory={isDirectory}
        onRowClick={() => { onContentClick(content);}}
        style={{
          marginLeft: '5vw',
          maxHeight: '50vh',
          maxWidth: '50vw'
        }}
      />
    );
  };

  renderLibraryItems = (contents) => contents.map(this.renderLibraryItem);

  renderTableBody = (contents) => (
    <TableBody>
      {this.renderLibraryItems(contents)}
    </TableBody>
  );

  render() {
    const { contents } = this.props;
    return (
      <Table
        className="library-menu"
        selectable="false"
      >
        <TableHead>
          <LibraryTableHead />
        </TableHead>
        {this.renderTableBody(contents)}
      </Table>
    );
  }
}

LibraryTable.propTypes = {
  contents: PropTypes.arrayOf(PropTypes.object).isRequired,
  onContentClick: PropTypes.func.isRequired
};

export default LibraryTable;
