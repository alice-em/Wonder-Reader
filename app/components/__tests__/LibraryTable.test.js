import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import LibraryItem from '../LibraryItem';
import LibraryTable, { generateLibraryItem, HeaderRow } from '../LibraryTable';

Enzyme.configure({ adapter: new Adapter() });

const generateContents = i => ({
  basename: `basename${i}`,
  dirname: `dirname${i}`,
  id: `id${i}`,
  isDirectory: i % 2 === 0,
});

const contents = [...Array(4).keys()].map(generateContents);

describe('LibraryTable', () => {
  it('should render', () => {
    const props = {
      contents,
      onContentClick: jest.fn(),
    };
    const wrapper = shallow(<LibraryTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('generateLibraryItem', () => {
    it('should call onContentClick', () => {
      const onContentClick = jest.fn();
      const content = {
        basename: 'basename',
        dirname: 'dirname',
        id: 'sampleId',
        isDirectory: true,
      };
      const Wrapper = generateLibraryItem(onContentClick);
      const wrapper = shallow(<Wrapper {...content} />);
      wrapper
        .find(LibraryItem)
        .props()
        .onRowClick.call();
      expect(onContentClick).toHaveBeenCalledWith(content);
    });
  });

  describe('HeaderRow', () => {
    it('should render', () => {
      const wrapper = shallow(<HeaderRow />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
