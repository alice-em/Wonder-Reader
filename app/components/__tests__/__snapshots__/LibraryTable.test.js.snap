// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryTable should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LibraryTable
    contents={
      Array [
        Object {
          "basename": "basename0",
          "dirname": "dirname0",
          "id": "id0",
          "isDirectory": true,
        },
        Object {
          "basename": "basename1",
          "dirname": "dirname1",
          "id": "id1",
          "isDirectory": false,
        },
        Object {
          "basename": "basename2",
          "dirname": "dirname2",
          "id": "id2",
          "isDirectory": true,
        },
        Object {
          "basename": "basename3",
          "dirname": "dirname3",
          "id": "id3",
          "isDirectory": false,
        },
      ]
    }
    onContentClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <HeaderRow />,
        <WithStyles(TableBody)>
          <LibraryItem
            basename="basename0"
            dirname="dirname0"
            id="id0"
            isDirectory={true}
            onRowClick={[Function]}
            style={
              Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              }
            }
          />
          <LibraryItem
            basename="basename1"
            dirname="dirname1"
            id="id1"
            isDirectory={false}
            onRowClick={[Function]}
            style={
              Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              }
            }
          />
          <LibraryItem
            basename="basename2"
            dirname="dirname2"
            id="id2"
            isDirectory={true}
            onRowClick={[Function]}
            style={
              Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              }
            }
          />
          <LibraryItem
            basename="basename3"
            dirname="dirname3"
            id="id3"
            isDirectory={false}
            onRowClick={[Function]}
            style={
              Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              }
            }
          />
        </WithStyles(TableBody)>,
      ],
      "className": "library-menu",
      "selectable": "false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <LibraryItem
              basename="basename0"
              dirname="dirname0"
              id="id0"
              isDirectory={true}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />,
            <LibraryItem
              basename="basename1"
              dirname="dirname1"
              id="id1"
              isDirectory={false}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />,
            <LibraryItem
              basename="basename2"
              dirname="dirname2"
              id="id2"
              isDirectory={true}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />,
            <LibraryItem
              basename="basename3"
              dirname="dirname3"
              id="id3"
              isDirectory={false}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "id0",
            "nodeType": "function",
            "props": Object {
              "basename": "basename0",
              "dirname": "dirname0",
              "id": "id0",
              "isDirectory": true,
              "onRowClick": [Function],
              "style": Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "id1",
            "nodeType": "function",
            "props": Object {
              "basename": "basename1",
              "dirname": "dirname1",
              "id": "id1",
              "isDirectory": false,
              "onRowClick": [Function],
              "style": Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "id2",
            "nodeType": "function",
            "props": Object {
              "basename": "basename2",
              "dirname": "dirname2",
              "id": "id2",
              "isDirectory": true,
              "onRowClick": [Function],
              "style": Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "id3",
            "nodeType": "function",
            "props": Object {
              "basename": "basename3",
              "dirname": "dirname3",
              "id": "id3",
              "isDirectory": false,
              "onRowClick": [Function],
              "style": Object {
                "marginLeft": "5vw",
                "maxHeight": "50vh",
                "maxWidth": "50vw",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <HeaderRow />,
          <WithStyles(TableBody)>
            <LibraryItem
              basename="basename0"
              dirname="dirname0"
              id="id0"
              isDirectory={true}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />
            <LibraryItem
              basename="basename1"
              dirname="dirname1"
              id="id1"
              isDirectory={false}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />
            <LibraryItem
              basename="basename2"
              dirname="dirname2"
              id="id2"
              isDirectory={true}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />
            <LibraryItem
              basename="basename3"
              dirname="dirname3"
              id="id3"
              isDirectory={false}
              onRowClick={[Function]}
              style={
                Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                }
              }
            />
          </WithStyles(TableBody)>,
        ],
        "className": "library-menu",
        "selectable": "false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <LibraryItem
                basename="basename0"
                dirname="dirname0"
                id="id0"
                isDirectory={true}
                onRowClick={[Function]}
                style={
                  Object {
                    "marginLeft": "5vw",
                    "maxHeight": "50vh",
                    "maxWidth": "50vw",
                  }
                }
              />,
              <LibraryItem
                basename="basename1"
                dirname="dirname1"
                id="id1"
                isDirectory={false}
                onRowClick={[Function]}
                style={
                  Object {
                    "marginLeft": "5vw",
                    "maxHeight": "50vh",
                    "maxWidth": "50vw",
                  }
                }
              />,
              <LibraryItem
                basename="basename2"
                dirname="dirname2"
                id="id2"
                isDirectory={true}
                onRowClick={[Function]}
                style={
                  Object {
                    "marginLeft": "5vw",
                    "maxHeight": "50vh",
                    "maxWidth": "50vw",
                  }
                }
              />,
              <LibraryItem
                basename="basename3"
                dirname="dirname3"
                id="id3"
                isDirectory={false}
                onRowClick={[Function]}
                style={
                  Object {
                    "marginLeft": "5vw",
                    "maxHeight": "50vh",
                    "maxWidth": "50vw",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "id0",
              "nodeType": "function",
              "props": Object {
                "basename": "basename0",
                "dirname": "dirname0",
                "id": "id0",
                "isDirectory": true,
                "onRowClick": [Function],
                "style": Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "id1",
              "nodeType": "function",
              "props": Object {
                "basename": "basename1",
                "dirname": "dirname1",
                "id": "id1",
                "isDirectory": false,
                "onRowClick": [Function],
                "style": Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "id2",
              "nodeType": "function",
              "props": Object {
                "basename": "basename2",
                "dirname": "dirname2",
                "id": "id2",
                "isDirectory": true,
                "onRowClick": [Function],
                "style": Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "id3",
              "nodeType": "function",
              "props": Object {
                "basename": "basename3",
                "dirname": "dirname3",
                "id": "id3",
                "isDirectory": false,
                "onRowClick": [Function],
                "style": Object {
                  "marginLeft": "5vw",
                  "maxHeight": "50vh",
                  "maxWidth": "50vw",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
